src/realm/obj.cpp

```diff
@@ -761,10 +770,87 @@ void out_mixed(std::ostream& out, const Mixed& val)
     }
 }
 
+void out_mixed_xjson(std::ostream& out, const Mixed& val)
+{
+    if (val.is_null()) {
+        out << "null";
+        return;
+    }
+    switch (val.get_type()) {
+        case type_Int:
+            out << "{\"$numberLong\": \"";
+            out << val.get<Int>();
+            out << "\"}";
+            break;
+        case type_Bool:
+            out << (val.get<bool>() ? "true" : "false");
+            break;
+        case type_Float:
+            out << "{\"$numberDouble\": \"";
+            out_floats<float>(out, val.get<float>());
+            out << "\"}";
+            break;
+        case type_Double:
+            out << "{\"$numberDouble\": \"";
+            out_floats<double>(out, val.get<double>());
+            out << "\"}";
+            break;
+        case type_String: {
+            out << "\"";
+            out_string(out, val.get<String>());
+            out << "\"";
+            break;
+        }
+        case type_Binary: {
+            out << "{\"$binary\": {\"base64\": \"";
+            out_binary(out, val.get<Binary>());
+            out << "\", \"subType\": \"00\"}}";
+            break;
+        }
+        case type_Timestamp: {
+            out << "{\"$date\": {\"$numberLong\": \"";
+            auto ts = val.get<Timestamp>();
+            int64_t timeMillis = ts.get_seconds() * 1000 + ts.get_nanoseconds() / 1000000;
+            out << timeMillis;
+            out << "\"}}";
+            break;
+        }
+        case type_Decimal:
+            out << "{\"$numberDecimal\": \"";
+            out << val.get<Decimal128>();
+            out << "\"}";
+            break;
+        case type_ObjectId:
+            out << "{\"$oid\": \"";
+            out << val.get<ObjectId>();
+            out << "\"}";
+            break;
+        case type_Link:
```

Comment:
For "link" and "list of link" types, even though they don't have a directly meaningful representation in extended json, I think we're going to need *some* way for those to be represented in the output so that we can actually use dumps that contain those types, since it will be entire possible for those to appear in the state files that get generated. Maybe you could just invent some pseudo-xjson representation for these? Like `{$realmLink: <primary key>}` or `{$realmLinkList: [ ... ] }` etc. Then the golang code that ingests the dump data can do some additional post-decode processing to specially handle those types.

Reply:
Oh, sorry. So these are just not handled as a part of out_mixed. Will tag you where these are handled